src/main.rs

```diff
@@ -50,23 +53,82 @@ mod errors {
         foreign_links {
             Io(::std::io::Error);
         }
+
+        errors {
+            NoCorrectStylesSpecified {
+                description("no correct styles specified")
+            }
+
+            UnknownStyleName(name: String) {
+                description("unknown style name")
+                display("unknown style name: '{}'", name)
+            }
+        }
     }
 }
 
 use errors::*;
 
-pub enum OptionsStyle {
-    Plain,
-    LineNumbers,
+#[derive(Debug, Eq, PartialEq)]
+enum OutputComponent {
+    Changes,
+    Grid,
+    Header,
+    Numbers,
+}
+
+impl FromStr for OutputComponent {
+    type Err = Error;
+
+    fn from_str(s: &str) -> Result<Self> {
+        match s {
+            "changes" => Ok(OutputComponent::Changes),
+            "grid" => Ok(OutputComponent::Grid),
+            "header" => Ok(OutputComponent::Header),
+            "numbers" => Ok(OutputComponent::Numbers),
+            _ => Err(ErrorKind::UnknownStyleName(s.to_owned()).into()),
+        }
+    }
+}
+
+#[derive(Debug, Eq, PartialEq)]
+enum PredefinedStyle {
     Full,
+    Plain,
+}
+
+impl PredefinedStyle {
+    fn value(&self) -> &'static [OutputComponent] {
+        match *self {
+            PredefinedStyle::Full => &[
+                OutputComponent::Changes,
+                OutputComponent::Grid,
+                OutputComponent::Header,
+                OutputComponent::Numbers,
+            ],
+            PredefinedStyle::Plain => &[],
+        }
+    }
+}
+
+impl FromStr for PredefinedStyle {
+    type Err = Error;
+
+    fn from_str(s: &str) -> Result<Self> {
+        match s {
+            "full" => Ok(PredefinedStyle::Full),
+            "plain" => Ok(PredefinedStyle::Plain),
+            _ => Err(ErrorKind::UnknownStyleName(s.to_owned()).into()),
```

Comment:
Should it fail or just default to full? The other options take this more forgiving style

Reply:
Defaulting to `full` happens in the clap-configuration: the default value for `--style` is `full`. Also, defaulting here would negate the selection-logic.